inst/include/Rcpp/exceptions.h

```diff
@@ -127,17 +127,17 @@ namespace Rcpp{
 } // namespace Rcpp
 
 inline SEXP get_last_call(){
-    SEXP sys_calls_symbol = Rf_install( "sys.calls" ) ;
-    Rcpp::Shield<SEXP> sys_calls_expr( Rf_lang1(sys_calls_symbol) );
+    SEXP sys_calls_symbol = Rf_install( "sys.call" ) ;
+
+    // -9 Skips the wrapped tryCatch from Rcpp_eval
+    Rcpp::Shield<SEXP> sys_calls_expr( Rf_lang2(sys_calls_symbol, Rf_ScalarInteger(-9)) );
```

Comment:
Yes the other way to do it would be to go back to using `sys.calls()` and return the call just prior to `tryCatch(sys.calls(), error = identity, interrupt = identity)`, which is what we call from the Rcpp_eval at https://github.com/RcppCore/Rcpp/pull/582/files#diff-5b6f3a71a6058dccfe3c3426c64110d2L132.

For example making a small change [1] to this PR, running the `tryCatch(takeLog-1), error = identity)` example from above, setting a breakpoint at exception.h:135 and running `Rf_PrintValue(calls)` gets you this callstack [2].

Writing the code to properly and robustly match `[[6]]` below is not something I have quite worked out how to do yet, which is why I went for the simpler solution to this PR initially, suggestions welcome.

[1]
```diff
diff --git a/inst/include/Rcpp/exceptions.h b/inst/include/Rcpp/exceptions.h
index b35d4bb..cd8e46f 100644
--- a/inst/include/Rcpp/exceptions.h
+++ b/inst/include/Rcpp/exceptions.h
@@ -127,10 +127,10 @@ namespace Rcpp{
 } // namespace Rcpp

 inline SEXP get_last_call(){
-    SEXP sys_calls_symbol = Rf_install( "sys.call" ) ;
+    SEXP sys_calls_symbol = Rf_install( "sys.calls" ) ;

     // -9 Skips the wrapped tryCatch from Rcpp_eval
-    Rcpp::Shield<SEXP> sys_calls_expr( Rf_lang2(sys_calls_symbol, Rf_ScalarInteger(-9)) );
+    Rcpp::Shield<SEXP> sys_calls_expr( Rf_lang1(sys_calls_symbol) );
     Rcpp::Shield<SEXP> calls( Rcpp_eval( sys_calls_expr, R_GlobalEnv ) );
     return calls;
 }
```

[2]
```r
[[1]]
tryCatch(takeLog(-1), error = identity)

[[2]]
tryCatchList(expr, classes, parentenv, handlers)

[[3]]
tryCatchOne(expr, names, parentenv, handlers[[1L]])

[[4]]
doTryCatch(return(expr), name, parentenv, handler)

[[5]]
takeLog(-1)

[[6]]
tryCatch(evalq(sys.calls(), <environment>), error = function (x)
x, interrupt = function (x)
x)

[[7]]
tryCatchList(expr, classes, parentenv, handlers)

[[8]]
tryCatchOne(tryCatchList(expr, names[-nh], parentenv, handlers[-nh]),
    names[nh], parentenv, handlers[[nh]])

[[9]]
doTryCatch(return(expr), name, parentenv, handler)

[[10]]
tryCatchList(expr, names[-nh], parentenv, handlers[-nh])

[[11]]
tryCatchOne(expr, names, parentenv, handlers[[1L]])

[[12]]
doTryCatch(return(expr), name, parentenv, handler)

[[13]]
evalq(sys.calls(), <environment>)

[[14]]
eval(substitute(expr), envir, enclos)
```